esm/src/api/controllers/demo.controller.js

```diff
@@ -0,0 +1,221 @@
+  try {
+    let doc = await deleteDemoItemService(req.params.demoId);
+    console.log(chalk.greenBright(`\n${item} DELETED successfully!\n`));
+    res.status(200).json({
+      message: `${item} deleted successfully!`,
+      request: {
+        type: "POST",
+        description: "Url link to make post request to",
+        url: `http://localhost:3000/${item}/`,
+        body: {
+          name: "String",
+          age: "Number",
+        },
+      },
+    });
+  } catch (err) {
+    res.status(500).json({
+      message: `Error deleting ${item}`,
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError deleting ${item}: ${err}\n`));
+  }
+};
+
+
+
+export {
+  getDemoItemsController,
+  createDemoItemController,
+  getOneDemoItemController,
+  deleteDemoItemController,
+};
+
+  
+  
+  
+  
+  
+  
+// router.patch("/:demoId", (req, res, next) => {
+//   const id = req.params.demoId;
+//   const updateOps = {};
+//   for (const ops of req.body) {
+//     updateOps[ops.propName] = ops.value;
+//   }
+//   Demo.updateOne({ _id: id }, { $set: updateOps })
+//     .exec()
+//     .then(() => {
+//       console.log(
+//         chalk.greenBright(
+//           `\nPATCH request for ID ${id} successful! \n\nUpdated ${item} url: http://localhost:3000/${routeName}/${id}\n`
+//         )
+//       );
+//       return res.status(200).json({
+//         message: "Patch request successful!",
+//         request: {
+//           type: "GET",
+//           description: `Url link to updated ${item}`,
+//           url: `http://localhost:3000/${routeName}/${id}`,
+//         },
+//       });
+//     })
+//     .catch((err) => {
+//       res.status(500).json({
+//         message: `Error updating ${item} property & value`,
+//         error: `${err}`,
+//       });
+//       console.log(
+//         chalk.redBright(`\nError updating ${item} property & value: ${err}\n`)
+//       );
+//     });
+// });
+
+// router.put("/:id", (req, res) => {
+//   let id = req.params.id;
+//   let resetItem = {
+//     name: req.body.name,
+//     age: req.body.age,
+//   };
+
+//   Demo.findByIdAndUpdate(id, { $set: resetItem }, { new: true })
+//     .exec()
+//     .then((response) => {
+//       console.log(
+//         chalk.greenBright(
+//           `\nPUT request for ID ${response._id} successful! \n\nUpdated ${item} url: http://localhost:3000/${routeName}/${id}\n`
+//         )
+//       );
+//       return res.status(200).json({
+//         message: `Put request successful!`,
+//         request: {
+//           type: "GET",
+//           description: `Url link to updated ${item}`,
+//           url: `http://localhost:3000/${routeName}/${id}`,
+//         },
+//       });
+//     })
+//     .catch((err) => {
+//       res.status(500).json({
+//         message: `Error updating ${item}`,
+//         error: `${err}`,
+//       });
+//       console.log(chalk.redBright(`\nError updating ${item}: ${err}\n`));
+//     });
+// });
```

Comment:
@samuko-things Nice work. I haven't tested that it works yet. But you should also restructure patch and put to mvc too.

Reply:
Also note: 
- You shouldn't remove the general checklist, add it back.
- The space before the #8 does not make it link your PR to the issue properly. Please link the issue